test/controllers/stocks_controller_test.rb

```diff
@@ -0,0 +1,40 @@
+require "test_helper"
+
+class StocksControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    @stock = create(:stock)
+    @user = create(:user)
+    sign_in @user
+  end
```

Comment:
It has been my experience that having test set up like this makes tests harder to write and maintain. In my opinion every test should have its own setup. DRY (Don't Repeat Yourself) doesn't apply to testing.